hamp_radar/flight_dataset.py

```diff
@@ -36,7 +36,13 @@ def read_datasetblock(
 def read_dataset(dsgeom: DatasetGeometry, postprocess: bool):
     ds_ppar = read_datasetblock(dsgeom.ppar)
 
-    dataset = [read_datasetblock(d, ds_ppar).chunk("auto") for d in dsgeom.data]
+    chunks = {
+        "cocx": 2,
+        "range": 512,  # TODO(ALL): see HACK about range dimension = 512
+        "fft": 256,
```

Comment:
These are likely good chunks for reading the pds, but we might want to bunch up a few timesteps when writing. Probably we also want to split at least co and cross channels (often only co is used) and we may want to split range.

We also need to ensure, that coordinate variables (especially 1D) are preferentially stored as a single chunk (e.g. time).

Reply:
- for splitting the co and cross channels and splitting range we can do that when writing really easily (I can add that)
- for bunching time I suggest we set the chunks of frame when reading to some value so that the overall chunk stays around 100MB, so maybe chunks `{"frame": 64, ...}` . Is that also what you had in mind?
- I'm not sure how to make coordinate variables be stored as a single chunk but I'll look into it

Reply:
Essentially, for writing, you'd want to define the chunking (and other things) in the `encoding` argument which you can pass to `to_zarr`. It's a dictionary containing individual settings for each variable.
The chunks here are `dask` chunks, which are in-memory chunks. It's best if the dask chunks align both, with the source data chunk pattern and the target chunk data pattern.

Reply:
I'm sorry I can't see how to put this into practise. I think I'm mistaken somewhere and it'd be better to discuss this in person, but here is what I understand so far:

- We then want all 1-D variables to be written as one chunk, which means they must already be one chunk when the dataset is created using dask. The 1-D variables had dim=frame so we need the frame dimension in the chunks dict when we concatenate the datasets to be at least the length of the concatenated dataset's frame dimension?? But this would have to be known before creating the concatenated dataset?? I don't think I understand this correctly.
- We also want to make coordinate variables stored as one chunk, but iq is a coordinate and is 4-D and massive, so making it one chunk seems like a very bad idea to me?
- I thought we want the zarr chunks to be around 10MB and so I thought we should make the frame chunk size the appropriate size such that when we make zarr chunks (with the co and cross channels and the range dimensions split) they would end up being about 10MB?

Reply:
Okay so based on what we discussed on Saturday I came up with a draft for how to chunk and re-chunk the dask arrays and then write them to smaller chunks in zarr. Let me know what you think! :)